File: src/array/remove_duplicated_sorted_array2/remove_duplicated_sorted_array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int count=1;
        int ptr = 1;
        int last = nums.length;
        while(ptr < last) {
            if (nums[ptr] == nums[ptr-1] && count < 2) {
                count++;
                ptr++;
            } else if (nums[ptr] == nums[ptr-1] && count == 2) {
                for(int j = ptr; j < last-1; j++) {
                    nums[j] = nums[j+1];
                }
                last--;
            } else {
                ptr++;
                count=1;
            }
        }
        return last;
    }
}

File: src/array/remove_duplicated_sorted_array2/remove_duplicated_sorted_array.py
"""

"""

class Solution:
    def removeDuplicates(self, nums: list) -> int:
        pt1 = 0
        pt2 = 1
        size = len(nums)
        count = 1
        k = 0
        while pt2 < size:
            num1 = nums[pt1]
            num2 = nums[pt2]
            if num1 == num2 and count == 2:
                nums.remove(num1)
                size -= 1
                continue
            elif num1 == num2 and count < 2:
                count += 1
            elif num1 != num2:
                count = 1
            pt1 += 1
            pt2 += 1

        return size 